### PR TITLE
Feat/add circleci pipeline

### LINKED ISSUE #6: Add circle ci pipeline

### PR DESCRIPTION
**Pull request by figuerascarlos**
This PR close #6

Automatics releases can be tackled in another PR.  

WDYT @fllaca  @hrodes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+jobs:
+  go-test:
+    executor:
+      name: go
+    steps:
+      - checkout
+      - run:
+          name: Install golangci-lint
+          command: |
+            curl -sLo golangci-lint.tar.gz https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz
```
Comment (hrodes) [inline .circleci/config.yml:20]: If we are going to download in every build may be it is better to just download and place it in the repo... Thoughts on this?

File: Makefile [hunk 2]
```diff
@@ -32,7 +32,7 @@ define compile
 endef
 
 .PHONY: build
-build: _bin lint test
+build: _bin lint
 	$(call compile, darwin)
```
Comment (hrodes) [inline Makefile:35]: should we add 32 bit support?

**Review by hrodes — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (fllaca) [reply .circleci/config.yml:20]: I would create our own version from `circleci/golang:1.14` with our custom tooling. This image should be public as well.

File: Makefile (hunk 2 above)
Comment (fllaca) [reply Makefile:35]: I think we can skip it for now.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            tar -xvf golangci-lint.tar.gz
+            mkdir -p ~/bin
+            mv golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint ~/bin
+            echo 'export PATH="~/bin:$PATH"'
+      - run:
+          name: Go verify
+          command: go mod verify
```
Comment (fllaca) [inline .circleci/config.yml]: should `go mod verify` part of `make test`?

**Review by fllaca — COMMENTED** [review]
lgtm, just minor thingies

File: Makefile (hunk 2 above)
Comment (figuerascarlos) [reply Makefile:35]: whats that of 32 bits ?? hahahah. 
 
for this milestone I don't think so, may we can add the issue and let that for future versions. WDYT ?

Comment (fllaca) [reply Makefile:35]: agree, for first version I think 64 bits should be enough

**Review by fllaca — APPROVED** [review]

**Pull request merged**